src/main/kotlin/no/nav/medlemskap/regler/v1/ReglerForMedl.kt

```diff
@@ -0,0 +1,159 @@
+package no.nav.medlemskap.regler.v1
+
+import no.nav.medlemskap.regler.common.*
+import no.nav.medlemskap.regler.common.Funksjoner.er
+
+class ReglerForMedl(val personfakta: Personfakta) : Regler() {
+
+
+    override fun hentHovedRegel(): Regel =
+            sjekkRegel {
+                harBrukerMedlOpplysninger
+            } hvisJa {
+                sjekkRegel {
+                    periodeMedOgUtenMedlemskap
+                } hvisJa {
+                    uavklartKonklusjon
+                } hvisNei {
+                    sjekkRegel {
+                        periodeMedMedlemskap
```

Comment:
Er de sjekkene du gjør her tilstrekkelige? Hvis jeg skjønner det korrekt, så sjekker du om det finnes en periode med medlemskap innenfor de 12 mnd vi er interessert i:
personfakta.personensPerioderIMedlSiste12Mnd().stream().anyMatch {
                    it.erMedlem && it.lovvalg er "NOR"
Og så sjekker du etterpå (i erPeriodeUtenMedlemskapInnenfor12MndPeriode) om det finnes en periode med ikke-medlemskap i den samme 12 mnd perioden?
Men vil du fange opp eventuelle hull i medlemskapet? Hvis personen ikke har registrert noen ikke-medlemskap, men bare har huller i medlemskapene? Ref https://confluence.adeo.no/pages/viewpage.action?pageId=370610913

Reply:
Hvordan er det med denne @olekvernberg ? Misforstår jeg helt her, eller har jeg et poeng?

Reply:
Du har som vanlig et godt poeng :) Det sjekkes på at en hel periode går over 12 mnd. Ser at navngivning kunne vært bedre, og skal endre det. Bruker må ha en medlemskapsperiode som går over hele 12 mnd-perioden. Svakheten med denne sjekken er at det kan gi uavklart på overlappende/connected medlemskapsperioder som egentlig er OK og som til sammen går over hele 12 mnd perioden. Det var egentlig bare et forsøk på å lage noe som er godt nok til å prodsette, og som er mulig å forbedre etterpå, for jeg synes denne PR'n allerede er litt for stor, og at denne branchen allerede har levd litt for lenge. På en annen side er det jo mange sammenhengende regler som må prodsettes samtidig, og at man dermed kan tillate litt voksne PR og at branchen lever lenge. Punktene jeg har listet opp i kommentaren til PR er også egentlig med dette i bakhodet, men det er mulig jeg har tenkt feil her, og egentlig burde bare konsentrere meg om å fullføre alt best mulig i branch, uten å presse meg på en rask merge. Innspill? :)

Reply:
Jeg tror man fint kan leve med at noen går til uavklart som ikke hadde trengt det, idag går jo alle med medl-perioder til uavklart, så alt vil være en forbedring. Så du kan fint prodsette det uten å gjøre noe mer med denne. Jeg hadde åpenbart ikke forstått koden helt korrekt, jeg skjønte ikke at det sjekkes at en hel periode går over 12 mnd, jeg trodde det bare ble sjekket startdatoen og sluttdatoen. Jeg er happy med dette her jeg altså, du kan ta resten senere.

Reply:
Det hadde vel vært verre hvis vi risikerte å gi ut JA-svar som burde vært uavklart..